src/pyodide/python-entrypoint-helper.ts

```diff
@@ -210,18 +210,29 @@ function makeEntrypointClass(className: string, classKind: AnyClass): any {
           if (typeof prop !== 'string') {
             return Reflect.get(target, prop, receiver);
           }
+          // Proxy calls to `fetch` to methods named `on_fetch` (and the same for other handlers.)
           const isKnownHandler = SUPPORTED_HANDLER_NAMES.includes(prop);
           if (isKnownHandler) {
             prop = 'on_' + prop;
           }
+
           return async function (...args: any[]): Promise<any> {
+            // Check if the requested method exists.
             const pyInstance = await target.pyInstance;
             if (typeof pyInstance[prop] === 'function') {
-              const res = await pyInstance[prop](...args);
-              if (isKnownHandler) {
-                return res?.js_object ?? res;
-              }
-              return res;
+              // Convert any JS arguments to Python SDK objects.
+              const pyodide = await getPyodide();
```

Comment:
Not sure I understand this, we said we're going to convert the parameters using decorators, why are we converting in the entrypoint-helper?

Reply:
We convert using decorator for top-level handlers, but we do it implicitly for entrypoint classes. I guess you could argue that this is inconsistent, I just really dislike having to write `@handler` everywhere, doing so for every RPC method would be annoying. Also these RPC methods aren't technically "handlers".

Reply:
This has the same problem that drove the push for the decorator
Users can’t modify this logic in their local workers-py module and use that instead
Its pre-baked into the runtime
I really strongly feel like we should minimize what the runtime bakes in without any ability for user overrides

Reply:
We can have the decorator for entry points be a class decorator or a meta class so that you wouldn't need to decorate every method

Reply:
Okay, if that's the concern then I think I see a way to do this in workers.py without the need for a decorator.

Reply:
I'd love to see the implementation but importing the workers.py from the user's bundle is not what I'd like to see either.
We should really keep the python entrypoint slim as possible.

I don't see the resistance to decorators, I really think this is the right call here.
We do the same for Rust Workers.

Reply:
> but importing the workers.py from the user's bundle is not what I'd like to see either.

Shouldn't be necessary if my idea pans out

> I don't see the resistance to decorators, I really think this is the right call here.

I think it's worthwhile to avoid as much clutter as possible when defining workers.

Reply:
Alright, curious to see what you come up with 😄

Reply:
Done! Essentially I moved the logic to workers.py, so each of our entrypoint classes implements it rather than it being implemented in the JS code.

Reply:
Oh! I like that!
Really clean and simple 🤩

Reply:
This makes me want default Class entrypoints even more.
Get away from the on_fetch handlers and move towards full class-based handlers that are auto-wrapped.

Reply:
And now Hood helped me make it more performant with some more advanced Python metaprogramming magic 🚀